neon/backends/kernels/maxas/MaxAs/MaxAsGrammar.pm

```diff
@@ -236,7 +236,7 @@ my $shfl  = qr"\.(?<mode>IDX|UP|DOWN|BFLY)";
 my $bar   = qr"\.(?<mode>SYNC|ARV|RED)(?:\.(?<red>POPC|AND|OR))? (?:$i8w4|$r8)(?:, (?:$i20w12|$r20))?(?(<r20>)|(?<nor20>))(?(<red>), $p39|(?<nop39>))"o;
 my $b2r   = qr"\.RESULT $r0(?:, $p45|(?<nop45>))"o;
 my $dbar  = qr"(?<SB>SB0|SB1|SB2|SB3|SB4|SB5)";
-my $dbar2 = qr" {(?<db5>5)?,?(?<db4>4)?,?(?<db3>3)?,?(?<db2>2)?,?(?<db1>1)?,?(?<db0>0)?}";
+my $dbar2 = qr" \{(?<db5>5)?,?(?<db4>4)?,?(?<db3>3)?,?(?<db2>2)?,?(?<db1>1)?,?(?<db0>0)?}";
```

Comment:
Hi @jamilbk Can you please explain why this "\" helps? 

Would this be compatible with lower Perl versions?

Reply:
I haven't tested it, but this was added to Perl 5.22 as a deprecation warning, and in 5.26 and 5.30 and later it is an error. So the `mnist_mkl.py` example is broken with recent versions of Perl and will be for all versions going forward.

Unfortunately I am not very proficient in Perl so the optimal fix here may be something else, but perhaps this will help:

https://stackoverflow.com/questions/31753737/deprecated-left-curly-bracket-in-perl-regex-exactly-when

Reply:
Ok, thanks for your link @jamilbk We'll take your PR for internal testing.